### PR TITLE
Better uses of Optional, inspired by the Java 11 upgrade

### PR DESCRIPTION
**Pull request by jmthibault79**
I thought it would be fun to play with some of the new Java 9-11 features.  No functional changes expected.

---
**PR checklist**

- [ ] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md)
- [ ] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md)
- [ ] If this PR is intended to complete a JIRA story, I have checked that all AC are met for that story
- [ ] I have manually tested this change and my testing process is described above
- [ ] This PR includes appropriate automated tests, and I have documented any behavior that cannot be tested with code
- [ ] If this fixes a bug, ensure the steps to reproduce are in the Jira ticket or provided above.
- [ ] I have added explanatory comments where the logic is not obvious
- [ ] If this change impacts deployment safety (e.g. removing/altering APIs which are in use) I have documented the impacts in the description
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P)
- [ ] If this includes an API change, I have run the relevant E2E tests locally because API changes are not covered by our PR checks

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/actionaudit/auditors/EgressEventAuditorImpl.kt [hunk 1]
```diff
@@ -114,7 +114,7 @@ constructor(
     private fun getDbWorkspace(event: SumologicEgressEvent): DbWorkspace {
         // Load the workspace via the GCP project name
         val dbWorkspaceMaybe = workspaceDao.getByGoogleProject(event.projectName)
-        if (!dbWorkspaceMaybe.isPresent()) {
+        if (dbWorkspaceMaybe.isEmpty) {
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/actionaudit/auditors/EgressEventAuditorImpl.kt:117]: Kotlin doesn't require the () - it's more idiomatic to exclude it, actually

File: api/src/main/java/org/pmiops/workbench/api/RuntimeController.java [hunk 2]
```diff
@@ -139,7 +139,7 @@ private String formatRuntimeErrors(@Nullable List<LeonardoClusterError> errors)
   private Runtime getOverrideFromListRuntimes(String googleProject) {
     Optional<LeonardoListRuntimeResponse> mostRecentRuntimeMaybe =
         leonardoNotebooksClient.listRuntimesByProject(googleProject, true).stream()
-            .sorted(
+            .min(
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/api/RuntimeController.java:142]: also new, as of Java 11

File: api/src/main/java/org/pmiops/workbench/iam/IamServiceImpl.java [hunk 3]
```diff
@@ -74,13 +74,10 @@ public List<String> revokeWorkflowRunnerRoleForUsers(
...
       for (String userEmail : userEmails) {
         // TODO can we make these requests in parallel?
-        Optional<String> petSaMaybe =
-            getOrCreatePetServiceAccountUsingImpersonation(googleProject, userEmail);
-        if (petSaMaybe.isPresent()) {
-          petServiceAccountsToRevokePermission.add(petSaMaybe.get());
-        } else {
-          petServiceAccountFailures.add(userEmail);
-        }
+        getOrCreatePetServiceAccountUsingImpersonation(googleProject, userEmail)
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/iam/IamServiceImpl.java:77]: not sure if this one is actually better

File: api/src/main/java/org/pmiops/workbench/access/AccessTierServiceImpl.java [hunk 4]
```diff
@@ -75,27 +75,26 @@ public void addUserToAllTiers(DbUser user) {
...
-      final DbUserAccessTier entryToInsert =
-          new DbUserAccessTier()
-              .setUser(user)
-              .setAccessTier(accessTier)
-              .setTierAccessStatus(TierAccessStatus.ENABLED)
-              .setFirstEnabled(now())
-              .setLastUpdated(now());
-      userAccessTierDao.save(entryToInsert);
-    }
+    userAccessTierDao
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/access/AccessTierServiceImpl.java:78]: is this better?

Comment (dmohs) [reply api/src/main/java/org/pmiops/workbench/access/AccessTierServiceImpl.java:78]: Yes, it is easier to follow the flow with this structure.

File: api/src/main/java/org/pmiops/workbench/iam/IamServiceImpl.java (hunk 3 above)
Comment (dmohs) [reply api/src/main/java/org/pmiops/workbench/iam/IamServiceImpl.java:77]: The concision here makes it easier to read.

**Review by dmohs — APPROVED** [review]
This is great. More concision is only beneficial to a point, but we are so far away from that line that all moves in that direction improve readability.

**Pull request merged**